pandas-stubs/core/indexes/period.pyi

```diff
@@ -52,6 +54,6 @@ def period_range(
     start: str | pd.Period | None = ...,
     end: str | pd.Period | None = ...,
     periods: int | None = ...,
-    freq: str | pd.DateOffset | None = ...,
+    freq: str | BaseOffset | None = ...,
```

Comment:
docs say `DateOffset` for `freq`, so concerned this could be too wide

Reply:
I tested with Day() which passes but I'd only BaseOffset not DateOffset.   I think this is a common flaw in the docs.

Reply:
What's not clear to me is whether there is anything that is a `BaseOffset` that is not a ` DateOffset`.  Looked at the pandas source and it does seem that `BaseOffset` is acceptable.

There is this code/comment in `offsets.pyx`:
```python
class OffsetMeta(type):
    """
    Metaclass that allows us to pretend that all BaseOffset subclasses
    inherit from DateOffset (which is needed for backward-compatibility).
    """
```

So I guess we're OK with your change here.